### PR TITLE
recording v2

### PR DESCRIPTION
**Pull request by frostbyte73**

### CODE REVIEW

File: recording/channels.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+func RequestChannel(id string) string {
+	return "RECORDING_REQUEST_" + id
+}
+
+func ResponseChannel(id string) string {
+	return "RECORDING_RESPONSE_" + id
+}
+
+func RecordingRPC(ctx context.Context, mb utils.MessageBus, recordingId string, req *livekit.RecordingRequest) error {
```
Comment (davidzhao) [inline recording/channels.go:29]: would you ever return a non-void response?

File: recording/channels.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		case <-time.After(RequestTimeout):
+			return errors.New("request timeout")
+		case msg := <-sub.Channel():
+			b = sub.Payload(msg)
+			resp := &livekit.RecordingResponse{}
+			err = proto.Unmarshal(b, resp)
+			if err != nil {
+				return err
+			}
+			if resp.RequestId != req.RequestId {
```
Comment (davidzhao) [inline recording/channels.go:56]: what if multiple of these are going and you are seeing a timing issue? this seem to suggest only one response could be in flight.

Comment (frostbyte73) [reply recording/channels.go:56]: if multiple are going, there will be multiple subscribers, and each will only care about their own request/response id

File: recording/channels.go (hunk 1 above)
Comment (frostbyte73) [reply recording/channels.go:29]: maybe in the future, but for now it's just an error

**Review by davidzhao — APPROVED** [review]
👍

**Pull request merged**